Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp

```diff
@@ -350,7 +354,22 @@ class CombinatorialKalmanFilter {
                            << tipState.nMeasurements
                            << ", nOutliers = " << tipState.nOutliers
                            << ", nHoles = " << tipState.nHoles << " on track");
-              result.trackTips.emplace_back(currentTip);
+              result.lastTrackIndices.emplace_back(currentTip);
+              // Set the lastMeasurementIndex to the last measurement
+              // to ignore the states after it in the rest of the algorithm
+              auto lastMeasurementIndex = currentTip;
+              bool isMeasurement =
+                  result.fittedStates.getTrackState(lastMeasurementIndex)
+                      .typeFlags()
+                      .test(TrackStateFlag::MeasurementFlag);
+              while (!isMeasurement) {
+                auto lastMeasurementState =
+                    result.fittedStates.getTrackState(lastMeasurementIndex);
+                lastMeasurementIndex = lastMeasurementState.previous();
+                isMeasurement = lastMeasurementState.typeFlags().test(
+                    TrackStateFlag::MeasurementFlag);
+              }
+              result.lastMeasurementIndices.emplace_back(lastMeasurementIndex);
```

Comment:
Do I understand correctly: 
 * sensitive surfaces, material surface can be found after the last measurement surface ?

Reply:
Yes the the condition to end the propagation through the detector is to reach the end of the detector. This mean that state are collected until the end even if the last measurement was past long ago. With this change we store two indices the one of the last state (in case someone want to use it) and the last measurement one. I had the same issue with the fitter in ATLAS : I would get 50 holes in the TRT even if my last measurement was in the SCT.